Neos.Neos/Classes/Domain/Link/Link.php

```diff
@@ -0,0 +1,173 @@
+ *
+ * (c) Contributors of the Neos Project - www.neos.io
+ *
+ * This package is Open Source Software. For the full copyright and license
+ * information, please view the LICENSE file which was distributed with this
+ * source code.
+ */
+
+declare(strict_types=1);
+
+namespace Neos\Neos\Domain\Link;
+
+use GuzzleHttp\Psr7\Uri;
+use Neos\Flow\Annotations as Flow;
+use Psr\Http\Message\UriInterface;
+
+/**
+ * Link value object modeled partially after the html spec for <a> tags.
+ *
+ * Note that currently the link editor can only handle and write to
+ * the property {@see Link::$target} "_blank" | null
+ * and to {@see Link::$rel} ["noopener"] | null
+ *
+ * The Link values can be accessed in Fusion the following:
+ *
+ * ```fusion
+ * href = ${q(node).property("link").href}
+ * title = ${q(node).property("link").title}
+ * # ...
+ * ```
+ *
+ * In case you need to cast the uri in {@see Link::$href} explicitly to a string
+ * you can use: `String.toString(link.href)`
+ *
+ * @Flow\Proxy(false)
+ */
+final readonly class Link implements \JsonSerializable
+{
+    /**
+     * A selection of frequently used target attribute values
+     */
+    public const TARGET_SELF = '_self';
+    public const TARGET_BLANK = '_blank';
+
+    /**
+     * A selection of frequently used rel attribute values
+     */
+    public const REL_NOOPENER = 'noopener';
+    public const REL_NOFOLLOW = 'nofollow';
+
+    /**
+     * @param array<int, string> $rel
+     */
+    private function __construct(
+        public UriInterface $href,
+        public ?string $title,
+        public ?string $target,
+        public array $rel,
+        public bool $download
+    ) {
+    }
+
+    /**
+     * @param array<int, string> $rel
+     */
+    public static function create(
+        UriInterface $href,
+        ?string $title,
+        ?string $target,
+        array $rel,
+        bool $download,
+    ): self {
+        $relMap = [];
+        foreach ($rel as $value) {
+            $relMap[trim(strtolower($value))] = true;
+        }
+        $trimmedTarget = $target !== null ? trim($target) : '';
+        return new self(
+            $href,
+            $title,
+            $trimmedTarget !== '' ? strtolower($trimmedTarget) : null,
+            array_keys($relMap),
+            $download
+        );
+    }
+
+    /**
+     * @param array<string, mixed> $array
+     */
+    public static function fromArray(array $array): self
+    {
+        return self::create(
+            new Uri($array['href']),
+            $array['title'] ?? null,
+            $array['target'] ?? null,
+            $array['rel'] ?? [],
+            $array['download'] ?? false,
+        );
+    }
+
+    public static function fromString(string $string): self
+    {
+        return self::create(
+            new Uri($string),
+            null,
+            null,
+            [],
+            false,
+        );
+    }
+
+    public function withTitle(?string $title): self
```

Comment:
we could compress these into a single wither to be able to set all four secondary members at once, wdyt?